inst/misc/develop_polars.R

```diff
@@ -0,0 +1,278 @@
+    ),
+    list(...),
+    .packages = list(.packages)
+  )
+
+  do.call(with_polars, c(list(\() system("cd ..; R CMD INSTALL --preclean --no-multiarch --with-keep.source r-polars")), args))
+
+}
+
+#' check polars with environment variables, filter errors, symlink  precompiled target
+#'
+#' @param RPOLARS_RUST_SOURCE where check can find precomiled target, set to '' if not use,
+#' DEFAULT is `paste0(getwd(),"/src/rust")`.
+#' @param ALL_FEATURES bool, to compile with e.g. simd, only for nightly toolchain
+#' @param NOT_CRAN bool, do not delete compiled target objects. Next compilation much faster.
+#' @param RPOLARS_CARGO_CLEAN_DEPS bool, do clean up cargo cache. Slows down next compilation
+#' @param ... other environment args to add
+#' @details in general Makevars check if bool-like envvars are not 'true'.
+#'
+#' @return no return
+#'
+#' @examples load_polars(ALL_FEATURES = '', SOME_OTHER_ENVVAR = 'true')
+check_polars = function(
+  RPOLARS_RUST_SOURCE = paste0(getwd(),"/src/rust"),
+  RPOLARS_ALL_FEATURES = 'true',
+  NOT_CRAN = 'true',
+  RPOLARS_CARGO_CLEAN_DEPS = 'false',
+  FILTER_CHECK_NO_FILTER = 'false',
+  ...,
+  check_dir = "./check/",
+  .packages = character()
+) {
+
+  # bundle all envvars
+  envvars = c(
+    list(
+      RPOLARS_RUST_SOURCE = RPOLARS_RUST_SOURCE,
+      RPOLARS_ALL_FEATURES = RPOLARS_ALL_FEATURES,
+      NOT_CRAN = NOT_CRAN,
+      RPOLARS_CARGO_CLEAN_DEPS = RPOLARS_CARGO_CLEAN_DEPS
+    ),
+    list(...)
+  )
+  not_cran = identical(NOT_CRAN,'true')
+  cat("check in not_cran mode:", not_cran ,"\n")
+  with_polars(
+    \() {
+      devtools::check(
+        env_vars = envvars, check_dir = check_dir,
+        vignettes = FALSE, cran = !not_cran
+      )
+      rextendr::document() # restore nanoarrow.Rd
+    },
+    RPOLARS_ALL_FEATURES = RPOLARS_ALL_FEATURES,
+    NOT_CRAN = NOT_CRAN,
+    RPOLARS_CARGO_CLEAN_DEPS = RPOLARS_CARGO_CLEAN_DEPS,
+    RPOLARS_RUST_SOURCE = RPOLARS_RUST_SOURCE,
+    ...
+  )
+
+  on.exit({unlink("check",recursive = TRUE)})
+  source("./inst/misc/filter_rcmdcheck.R")
+  print("check polars is done")
+}
+
+#' submit polars with environment variables, filter errors, symlink  precompiled target
+#'
+#' @param RPOLARS_RUST_SOURCE where check can find precomiled target, set to '' if not use,
+#' DEFAULT is `paste0(getwd(),"/src/rust")`.
+#' @param ALL_FEATURES bool, to compile with e.g. simd, only for nightly toolchain
+#' @param NOT_CRAN bool, do not delete compiled target objects. Next compilation much faster.
+#' @param RPOLARS_CARGO_CLEAN_DEPS bool, do clean up cargo cache. Slows down next compilation
+#' @param ... other environment args to add
+#' @details in general Makevars check if bool-like envvars are not 'true'.
+#'
+#' @return no return
+#'
+#' @examples load_polars(ALL_FEATURES = '', SOME_OTHER_ENVVAR = 'true')
+submit_polars = function(
+  RPOLARS_RUST_SOURCE = paste0(getwd(),"/src/rust"),
+  RPOLARS_ALL_FEATURES = 'false',
+  NOT_CRAN = 'true',
```

Comment:
Do not package with `NOT_CRAN="true"`.

Reply:
hmm could we make some envvar called REUSE_BUT_DO_NOT_DELETE_CACHE ? The compilation times are really killing it for me. I have spend 7 almost full work days on this CRAN submission alone.

Reply:
If I understand correctly, the packaging for submission to CRAN is done in a temporary directory, so I don't think the cache used during development will be deleted.

And, unfortunately, I think deleting the cache is a requirement for builds on CRAN. See this.
https://github.com/r-rust/hellorust/blob/551df61f2dd8069ac3ef4da6a65f8f0338b67b2f/src/Makevars#L16-L19